src/transformers/models/character_bert/modeling_character_bert.py

```diff
@@ -0,0 +1,1938 @@
+
+    def forward(self, inputs: torch.Tensor) -> torch.Tensor:
+        current_input = inputs
+        for layer in self._layers:
+            projected_input = layer(current_input)
+            linear_part = current_input
+            # NOTE: if you modify this, think about whether you should modify the initialization
+            # above, too.
+            nonlinear_part, gate = projected_input.chunk(2, dim=-1)
+            nonlinear_part = self._activation(nonlinear_part)
+            gate = torch.sigmoid(gate)
+            current_input = gate * linear_part + (1 - gate) * nonlinear_part
+        return current_input
+
+
+class CharacterCnn(torch.nn.Module):
+    """
+    NOTE: Adapted from _ElmoCharacterEncoder:
+    https://github.com/allenai/allennlp/blob/main/allennlp/modules/elmo.py#L254
+    Computes context insensitive token representation using multiple CNNs.
+    This embedder has input character ids of size (batch_size, sequence_length, 50)
+    and returns (batch_size, sequence_length, hidden_size), where hidden_size
+    is typically 768.
+    """
+    def __init__(self, config):
+        super().__init__()
+        self.character_embeddings_dim = config.character_embeddings_dim
+        self.cnn_activation = config.cnn_activation
+        self.cnn_filters = config.cnn_filters
+        self.num_highway_layers = config.num_highway_layers
+        self.max_word_length = config.max_word_length
+        self.hidden_size = config.hidden_size
+        # NOTE: this is the 256 possible utf-8 bytes + special slots for the
+        # [CLS]/[SEP]/[PAD]/[MASK] characters as well as beginning/end of
+        # word symbols and character padding for short words -> total of 263
+        self.character_vocab_size = 263
+        self._init_weights()
+
+    def get_output_dim(self):
+        return self.hidden_size
+
+    def _init_weights(self):
+        self._init_char_embedding()
+        self._init_cnn_weights()
+        self._init_highway()
+        self._init_projection()
+
+    def _init_char_embedding(self):
+        weights = torch.empty((
+            self.character_vocab_size,
+            self.character_embeddings_dim))
+        nn.init.normal_(weights)
+        weights[0].fill_(0.)  # token padding
+        weights[CharacterMapper.padding_character + 1].fill_(0.)  # character padding
+        self._char_embedding_weights = torch.nn.Parameter(
+            torch.FloatTensor(weights),
+            requires_grad=True
+        )
+
+    def _init_cnn_weights(self):
+        convolutions = []
+        for i, (width, num) in enumerate(self.cnn_filters):
+            conv = torch.nn.Conv1d(
+                in_channels=self.character_embeddings_dim,
+                out_channels=num, kernel_size=width, bias=True)
+            conv.weight.requires_grad = True
+            conv.bias.requires_grad = True
+            convolutions.append(conv)
+            self.add_module("char_conv_{}".format(i), conv)
```

Comment:
```suggestion
            self.add_module(f"char_conv_{i}", conv)
```
We only use f-strings in the library.

Reply:
Noted.